src/main/scala/mesosphere/marathon/core/autoscale/impl/AutoScaleAppActor.scala

```diff
@@ -0,0 +1,127 @@
+package mesosphere.marathon.core.autoscale.impl
+
+import akka.actor._
+import akka.pattern.pipe
+import mesosphere.marathon.core.autoscale.impl.AutoScaleActor.{ AutoScaleFailure, AutoScaleSuccess }
+import mesosphere.marathon.core.autoscale.impl.AutoScaleAppActor._
+import mesosphere.marathon.core.autoscale.{ AutoScaleConfig, AutoScalePolicy, AutoScaleResult }
+import mesosphere.marathon.core.task.tracker.TaskTracker
+import mesosphere.marathon.state.{ AppDefinition, GroupManagement, Timestamp }
+import mesosphere.marathon.upgrade.{ DeploymentPlan, DeploymentStep, ScaleApplication }
+import mesosphere.marathon.{ ConflictingChangeException, MarathonSchedulerService }
+import mesosphere.util.ThreadPoolContext
+
+import scala.concurrent.Future
+
+object AutoScaleAppActor {
+
+  sealed trait AutoScaleActionResult
+  case object NoActionNecessary extends AutoScaleActionResult
+  case class DeploymentTriggered(result: AutoScaleResult, deployment: DeploymentPlan) extends AutoScaleActionResult
+  case class DeploymentBlocked(result: AutoScaleResult) extends AutoScaleActionResult
+
+  def props(app: AppDefinition,
+            lastFailure: Option[Timestamp],
+            handler: ActorRef,
+            policies: Seq[AutoScalePolicy],
+            groupManager: GroupManagement,
+            scheduler: MarathonSchedulerService,
+            taskTracker: TaskTracker,
+            conf: AutoScaleConfig): Props = {
+    Props(new AutoScaleAppActor(app, lastFailure, handler, policies, groupManager, scheduler, taskTracker, conf))
+  }
+}
+
+class AutoScaleAppActor(app: AppDefinition,
```

Comment:
Why is this an actor and not just a service? It doesn't have any state, does it?